internal/replication/blockreplicator_3node_test.go

```diff
@@ -309,3 +310,99 @@ func TestBlockReplicator_3Node_SubmitRecover(t *testing.T) {
 		require.NoError(t, err)
 	}
 }
+
+// Scenario:
+// - Configure cluster to take snapshots every approx. 5 blocks.
+// - Start 3 nodes together, wait for leader, submit 10 blocks, wait for all ledgers to get them.
+// - Stop a follower node,  wait for leader, submit 10 blocks, wait for 2 ledgers to get them.
+// - Restart the node, wait for leader, wait for node to get missing blocks.
+//   Recovering node is expected to get a snapshot from the leader and trigger catch-up from it.
+// - Stop a leader node,  wait for new leader, submit 10 blocks, wait for 2 ledgers to get them.
+// - Restart the node, wait for leader, wait for node to get missing blocks.
+//   Recovering node is expected to get a snapshot from the leader and trigger catch-up from it.
+func TestBlockReplicator_3Node_Catchup(t *testing.T) {
```

Comment:
does not cover the case where the puller is not returning all blocks at once. Is it possible to cover that case without using mocks? I think it is possible by doing multiple sets of block commits and closing multiple followers, etc...

Reply:
Right now that would be very difficult and unreliable, likely to produce flakes.... but with the pull memory management feature planned here https://github.com/ibm-blockchain/bcdb-server/issues/189, it would be quite easy and reliable. I'll tackle it then, if you don't object.

Reply:
Agree